### ISSUE TITLE
"Multiple private IPv4 addresses found" even though consul validate passes

### ISSUE THREAD

**Issue by MrMMorris** [hashicorp/consul#15536]
#### Overview of the Issue

Recently upgraded from 0.6.0 to 1.14.0. When starting consul, I get the following error:

`==> Multiple private IPv4 addresses found. Please configure one with 'bind' and/or 'advertise'.`

Looked at multiple issues and posts about this and tried setting `bind_addr`, `advertise_addr` and `client_addr` to private IP or 0.0.0.0 in any combination. 

Now I am at a point where `consul validate` gives `Configuration is valid!` but running consul still gives the error 

### Operating system and Environment details

Consul v1.14.0 on Ubuntu 18.04

### Consul config

```
{
  "start_join": ["10.0.0.9", "10.0.1.6", "10.0.0.5"],
  "disable_remote_exec": true,
  "domain": "consul.",
  "data_dir": "/opt/consul/data",
  "dns_config": {
    "allow_stale": true,
    "max_stale": "5s",
    "node_ttl": "30m",
    "service_ttl": {
      "*": "0s"
    },
    "enable_truncate": false,    
    "only_passing": false
  },
  "log_file": "/var/log/consul",
  "log_level": "INFO",
  "log_rotate_bytes": 1000000,
  "log_rotate_max_files": 10,
  "node_name": "dev-1",
  "server": false,
  "client_addr": "10.0.1.58",
  "bind_addr": "0.0.0.0",
  "advertise_addr": "10.0.1.58",
  "datacenter": "ec2",
  "encrypt": "XXXXXXXXXX",
  "ports": {
    "dns": 8600,
    "http": 8500,
    "https": -1,
    "serf_lan": 8301,
    "serf_wan": 8302,
    "server": 8300
  },
  "rejoin_after_leave": true,
  "leave_on_terminate": true,
  "disable_update_check": false
}
```

#### Reproduction Steps

Using the above config with v1.14.0:
1. `consul validate -config-format=json /etc/consul.conf`
    `Configuration is valid!`
2. `/opt/consul/bin/consul agent -config-file=/etc/consul.conf`
    `==> Multiple private IPv4 addresses found. Please configure one with 'bind' and/or 'advertise'.`

**Comment by blake** [comment]
This error is raised when there are multiple IPv4 addresses present on the host system.

In order to resolve this error, change the value of [`-bind-`](https://developer.hashicorp.com/consul/docs/agent/config/cli-flags#_bind) or [`bind_addr`](https://developer.hashicorp.com/consul/docs/agent/config/config-files#bind_addr) from `0.0.0.0` to a specific IP address or go-sockaddr template that resolves to a single IP.

For example:

```json
{
  "bind_addr": "{{ GetPrivateInterfaces | include \"network\" \"192.0.2.0/24\" | attr \"address\" }}"
}
```

**Comment by MrMMorris** [comment]
thanks for the response, @blake

Unfortunately, I have tried that and receive the same error:

```
  "client_addr": "10.0.1.58",
  "bind_addr": "10.0.1.58",
  "advertise_addr": "10.0.1.58",
```


`/opt/consul/bin/consul agent -config-file=/etc/consul.conf`
`==> Multiple private IPv4 addresses found. Please configure one with 'bind' and/or 'advertise'.`

**Comment by MrMMorris** [comment]
never mind I think I figured it out! 

Seems that the error was due to using `consul.conf` instead of `consul.json` which meant my config wasn't even being used and so hence the error (I figured that out by specifying `-bind` when running instead of in the config so I could see the error). I then ran into [this issues](https://github.com/hashicorp/consul/issues/4961) which made me aware of your ansible playbook which I will be using instead of the other I had before.

thanks for indirectly helping me [🙏](https://emojipedia.org/folded-hands/)
